### PR TITLE
Add new item similarity metrics for SAR

### PR DESCRIPTION
**Pull request by simonzhaoms**
### Description

This PR adds 4 new item similarity metrics for SAR:
* Mutual information: $s_{ij} = \log_2\frac{nc_{ij}}{c_{ii}c_{jj}}$
* Lexicographers mutual information: $s_{ij} = c_{ij}\log_2\frac{nc_{ij}}{c_{ii}c_{jj}}$
* Cosine similarity: $s_{ij} = \frac{c_{ij}}{\sqrt{c_{ii}c_{jj}}}$
* Inclusion index: $s_{ij} = \frac{c_{ij}}{\min (c_{ii}, c_{jj})}$

### Related Issues

* https://github.com/microsoft/recommenders/issues/126

### Checklist:

- [X] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [X] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

**Comment by miguelgfierro** [comment]
Something to consider @simonzhaoms is a metric called [Earth Mover's Distance](https://en.wikipedia.org/wiki/Earth_mover%27s_distance#:~:text=In%20statistics%2C%20the%20earth%20mover%27s%20distance%20%28EMD%29%20is,mathematics%2C%20this%20is%20known%20as%20the%20Wasserstein%20metric.) I have used that in the past and worked well for me. It was for comparing feature vectors and I was using a large number of distances: euclidean, KL, [Jasen-Shannon](https://en.wikipedia.org/wiki/Jensen%E2%80%93Shannon_divergence), and others. And I remember that this one got really good results. Not sure if it could be applicable in the SAR scenario

File: recommenders/utils/python_utils.py [hunk 1]
```diff
@@ -63,16 +75,82 @@ def lift(cooccurrence):
...
 
     with np.errstate(invalid="ignore", divide="ignore"):
         result = cooccurrence / (diag_rows * diag_cols)
 
     return np.array(result)
 
 
+def mutual_information(cooccurrence):
+    """Helper method to calculate the Mutual Information of a matrix of co-occurrences.
+
```
Comment (miguelgfierro) [inline recommenders/utils/python_utils.py:88]: Can we add here what is the effect of all these new metrics? Maybe explain the difference between them as we have in jaccard, lift, etc https://github.com/microsoft/recommenders/blob/staging/recommenders/utils/python_utils.py#L30

Let @anargyri or me know if you need support with the explanations

Comment (simonzhaoms) [reply recommenders/utils/python_utils.py]: Added

**Review by miguelgfierro — APPROVED** [review]
This is really good and interesting work Simon

**Pull request merged**